### ISSUE TITLE
Allow record based naming strategies

### ISSUE THREAD

**Issue by FabianF92** [awslabs/aws-glue-schema-registry#17]
Hi,

we are currently migrating our registry to AWS Glue. 
There are some kafka topics in our project that do not have a specific avro schema attached, but they can receive messages serialized by different schemas.
Therefore, we would like to register those schemas not on a topic based naming strategy.
However, interface `AWSSchemaNamingStrategy` does only provide a method `getSchemaName(String transportName)`, that generates schema name based on the topic.

It would be great if the interface method could be extended by the schema parameter. This allows to implement strategies that are record based.
As far as I can see, `prepareInput` in `AWSKafkaAvroSerializer` already gets `data` as input, so one could use `AVROUtils` to extract schema and pass it to `getSchemaName`.

**Comment by mohitpali** [comment]
Thanks for reaching out to us.

We are looking into this and will update it in our upcoming releases.

**Comment by mohitpali** [comment]
Hi - We have merged a change into this repository. Could you please validate if it unblocks you ?

**Comment by FabianF92** [comment]
Hi, thank you. Yes it does!
When is the next release into maven repository planned?

**Comment by mohitpali** [comment]
We are planning a new incremental version release this month (2/2021).
I have moved your comment to a new issue(https://github.com/awslabs/aws-glue-schema-registry/issues/27) and will close this one.
Thanks.
